src/controller/event/handler/internal/artifact.go

```diff
@@ -240,3 +244,21 @@ func (a *Handler) onPush(ctx context.Context, event *event.ArtifactEvent) error
 
 	return nil
 }
+
+// isScannerUser check if the current user is a scanner user by its prefix
+// usually a scanner user should be named like `robot$<projectName>+<Scanner UUID (8byte)>-<Scanner Name>-<UUID>`
+// verify it by the prefix `robot$<projectName>+<Scanner UUID (8byte)>`
+func isScannerUser(ctx context.Context, operator string, projectID int64) bool {
```

Comment:
we can transmit the repository of into ArtifactEvent this function to avoid one DB call?

Reply:
Updated